### PR TITLE
Ensuring empty runtime versions are not used when deploying.

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
#### Description
Ensuring we always have a runtime version string when launching a deployment. Previously using an empty runtime version would result in a vague error.

Also changed up how the string is retrieved for the deployment options. The value can be empty which reflects the settings in editor settings. If "Use GDK Pinned version" is enable the string will display the pinned version.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp [hunk 1]
```diff
@@ -720,5 +720,6 @@ bool SSpatialGDKSimulatedPlayerDeployment::IsUsingCustomRuntimeVersion() const
 FText SSpatialGDKSimulatedPlayerDeployment::GetSpatialOSRuntimeVersionToUseText() const
 {
 	const USpatialGDKEditorSettings* SpatialGDKSettings = GetDefault<USpatialGDKEditorSettings>();
-	return FText::FromString(SpatialGDKSettings->GetSpatialOSRuntimeVersionForCloud());
+	FString RuntimeVersion = SpatialGDKSettings->bUseGDKPinnedRuntimeVersion ? SpatialGDKServicesConstants::SpatialOSRuntimePinnedVersion : SpatialGDKSettings->CloudRuntimeVersion;
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp:723]: Why this change ? This was intended to show information that's hidden in code, but now it displays what the user entered in a field just below.

Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp:723]: Just checking, if the user set an empty string in the setting, this will display an empty string as well. But that's OK because the configuration is generated from the settings anyway, and we will loopback through GetSpatialOSRuntimeVersionForCloud ?

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp:723]: Yup exactly. The runtime version that will be used would in the case of an input empty string would be the pinned version.
